### PR TITLE
cylc-rose: `cylc list` compatibility

### PR DESCRIPTION
**Pull request by wxtim**
These changes address #4288 
These changes are twinned with https://github.com/cylc/cylc-rose/pull/65

Make the following scripts work with cylc-rose:

### Internal Todo List

- [x] Cylc list
- [x] Cylc graph
- [x] Cylc config
- [x] Cylc validate

### Standard checklist

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.py` and
  `conda-environment.yml`.
- [x] Does not need tests (tests are in cylc-rose).
- [x] Appropriate change log entry included:  https://github.com/wxtim/cylc/pull/30 (Can be merged before merging this PR, avoids conflict)
- [x] Documented changes in cylc list docstring: Auto documents using the Cylc Rose args.

### How to review:

Create workflows with a `rose-suite.conf` in
- `~/cylc-src/`
-  `~/cylc-run/` (using `cylc install`)
- some other folder somewhere else entirely

You might use jinja2 or empy in the workflow to carry out tests like [these](https://github.com/cylc/cylc-rose/blob/master/tests/functional/06_jinja2_thorough/flow.cylc).

Try each of the four cylc CLI commands with and without the [extra rose options provided by this function](https://github.com/wxtim/cylc/blob/5ec95b29c20fe74a49754f529317f5f10597ab6d/cylc/flow/scripts/install.py#L86)/

Ideally these commands and `cylc install` should behave in a consistent way.

### CODE REVIEW

**Comment by wxtim** [comment]
File containing a list of tests which need to be rub locally with https://github.com/cylc/cylc-rose/pull/65 installed.

[fail_online.txt](https://github.com/cylc/cylc-flow/files/6800099/fail_online.txt)

**Comment by kinow** [comment]
> File containing a list of tests which need to be run locally with [cylc/cylc-rose#65](https://github.com/cylc/cylc-rose/pull/65) installed: These will fail on GH Actions even if this PR is OK.
> 
> [fail_online.txt](https://github.com/cylc/cylc-flow/files/6800099/fail_online.txt)

I think you fixed this right? (In GH UI, we can mark a comment like this as "Outdated" and hide it using the three dots at the top right corner of the comment area)

**Review by kinow — COMMENTED** [review]
Code and tests look good! Installing everything (flow + rose from both new branches) to test it now.

File: cylc/flow/scripts/install.py [hunk 1]
```diff
@@ -83,6 +83,51 @@
...
+
+    Args:
+        parser: An option parser object
+    """
+    try:
+        __import__('cylc.rose')
+        parser.add_option(
+            "--opt-conf-key", "-O",
+            help=(
+                "Use optional Rose Config Setting"
```
Comment (kinow) [inline cylc/flow/scripts/install.py:97]: Missing space here?

image

**Review by kinow — COMMENTED** [review]
@wxtim I used @oliver-sanders ' example from the linked issue.

1. Created `~/cylc-src/rose1/flow.cylc` with the following content:

```
[scheduling]
    initial cycle point = {{ ICP }}
    [[graph]]
        R1 = foo
```

Then created a `~/cylc-src/rose1/rose-suite.conf` (not using `myopt` as in his example for now, will after this simpler example works).

```
[template variables]

ICP = "20000101T0000Z"
```

3. Installed the workflow

```
(venv) kinow@ranma:~/cylc-src/rose1$ cylc install
INSTALLED rose1/run1 from /home/kinow/cylc-src/rose1
```

4. Verified the files were installed/copied correctly

```
(venv) kinow@ranma:~/cylc-src/rose1$ ls ~/cylc-run/rose1/run1/
flow.cylc  log  opt  rose-suite.conf
```

5. Tried `cylc validate rose1/run1`

```
(venv) kinow@ranma:~/cylc-src/rose1$ cylc validate rose1/run1
IllegalValueError: (type=cycle point) [scheduling]initial cycle point = {{ ICP }}
```

Any idea what step I missed? I am trying @oliver-sanders ' example first as that was in the ticket, and looks like a simple way to test this PR. Will try more Jinja2/empy once I get this one working :)

Thanks!!!
Bruno

File: cylc/flow/scripts/install.py (hunk 1 above)
Comment (wxtim) [reply cylc/flow/scripts/install.py:97]:
```suggestion
                "Use optional Rose Config Setting "
```

**Comment by wxtim** [comment]
> Any idea what step I missed? I am trying @oliver-sanders ' example first as that was in the ticket, and looks like a simple way to test this PR. Will try more Jinja2/empy once I get this one working :)

You might have not explicitly stated because it seems obvious, but I you haven't put a `#!jinja2` line in your `flow.cylc`: This is fine if you are using the Rose 1 legacy `rose-suite.conf:[jinja2:suite.rc]` section heading, but with `[template variables]` there is no way way Cylc-Rose can deduce the templating language.

**Comment by kinow** [comment]
Ah! You are right @wxtim ! Added the shebang, and also the `runtime` section, now it's working fine.

```
#!jinja2

[scheduling]
    initial cycle point = {{ ICP }}
    [[graph]]
        R1 = foo

[runtime]
[[foo]]

```

```
(venv) kinow@ranma:~/cylc-src/rose1$ cylc install 
INSTALLED rose1/run1 from /home/kinow/cylc-src/rose1
(venv) kinow@ranma:~/cylc-src/rose1$ cylc validate rose1/run1
Valid for cylc-8.0b2.dev
```

Will finish the review tomorrow morning. Thanks again!! :+1:

**Review by kinow — APPROVED** [review]
Tested the other commands, `list`, `config`, and `graph`. Everything worked fine. Changed the workflow to be:

```
#!empy

[scheduling]
    initial cycle point = @ICP
    [[graph]]
        R1 = foo

[runtime]
[[foo]]
```

And confirmed I had `empty` installed. Everything worked fine too :-) Great job @wxtim !

p.s.: we probably have a ticket somewhere to rename `rose-suite.conf` to `rose-workflow.conf` I think?

**Comment by wxtim** [comment]
> p.s.: we probably have a ticket somewhere to rename `rose-suite.conf` to `rose-workflow.conf` I think?

No. A rose suite is a config for a Cylc workflow. No immediate plans to change the term.

**Review by datamel — APPROVED** [review]
Thanks @wxtim 
Looks great. I have read the code, checked out the branch (with its sibling), ran the tests and the manual one as detailed above. I have also locally run the functional tests which were affected. No issues found for me.

**Comment by oliver-sanders** [comment]
@wxtim, just noticed this PR was marked as "partially addresses #4288", could you add a note to #4288 covering what's left for future work.

**Pull request merged**